src/goto-programs/goto_model.h

```diff
@@ -95,6 +97,27 @@ class goto_modelt : public abstract_goto_modelt
     return goto_functions.function_map.find(id) !=
            goto_functions.function_map.end();
   }
+
+  /// Iterates over the functions inside the goto model and checks invariants
+  /// in all of them. Prints out error message collected.
+  /// \param msg message instance to collect errors
+  /// \return true if any violation was found
+  bool check_internal_invariants(messaget &msg) const
+  {
+    bool found_violation = false;
+    namespacet ns(symbol_table);
+    forall_goto_functions(it, goto_functions)
+    {
+      if(!base_type_eq(
+           it->second.type, symbol_table.lookup_ref(it->first).type, ns))
+      {
+        msg.error() << "error" << messaget::eom;
```

Comment:
Say what's wrong!

Reply:
Yes, a better error message here would be good.